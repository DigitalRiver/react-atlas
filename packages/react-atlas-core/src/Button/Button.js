import React from "react";
import PropTypes from "prop-types";
import {IconCore} from './../Icon';
import cx from "classnames";

/**
 * A Generic button component.
 */
class Button extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      onClick,
      className,
      children,
      disabled,
      large,
      small,
      primary,
      secondary,
      warning,
      error,
      link,
      outline,
      icon,
      type,
      href,
      style
    } = this.props;

    let mainStyle = "button";
    if (primary) {
      mainStyle = "primary";
    } else if (secondary) {
      mainStyle = "secondary";
    } else if (warning) {
      mainStyle = "warning";
    } else if (error) {
      mainStyle = "error";
    } else if (link) {
      mainStyle = "link";
    } else {
      mainStyle = "default_btn";
    }

    if (outline) {
      mainStyle += "_outline";
    }

    /**
     * Use cx to construct a CSS class object.
     * cx will automatically remove falsy values
     * for us. This way only passed props will be
     * transformed into CSS classes.
     */
    const classes = cx(
      {
        "disabled": disabled,
        "large": large,
        "small": small,
        "button": true
      },
      mainStyle
    );

    let text = children;
    let iconClass = false;

    /* Just use the icon instead of text when an icon
     * is passed and no children(text) is set. */
    if (icon && children === "Default Button") {
      text = null;
    } else if (icon) {
      text = children;
      iconClass = "ra_Button__icon-left";
    }

    let btn = <button
          onClick={onClick}
          className={cx(className)}
          styleName={classes}
          style={style}
          type={type}
          href={href}
        >
          {icon ? <IconCore icon={icon} /> : null}
          {text}
        </button>
    ;

    let renderButton = btn;
    if (href) {
      renderButton = <a href={href}>{btn}</a>;
    }

    return renderButton;
  }
}

Button.propTypes = {
  /** HTML5 button type, eg submit, button, etc */
  "type": PropTypes.string,
  /** The URL to link to. */
  "href": PropTypes.string,
  /**
   * Define a mini button.
   *
   */
  "children": PropTypes.node,
  /**
   * Click event handler.
   */
  "onClick": PropTypes.func,
  /** An Object, array, or string of CSS classes to apply to button.*/
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),
  /**
   *Determines if the button is disabled.
   * @examples <Button disabled/>
   */
  "disabled": PropTypes.bool,
  /**
   * use outline styled button
   * @examples <Button outline>
   */
  "outline": PropTypes.bool,
  /**
   * use primary style button
   * @examples <Button primary>
   */
  "primary": PropTypes.bool,
  /**
   * use secondary style button
   * @examples <Button secondary>
   */
  "secondary": PropTypes.bool,
  /**
   * use warning style button
   * @examples <Button warning>
   */
  "warning": PropTypes.bool,
  /**
   * use error style button
   * @examples <Button error>
   */
  "error": PropTypes.bool,
  /**
   * use link style button
   * @examples <Button link>
   */
  "link": PropTypes.bool,
  /**
   * use large style button
   * @examples <Button large>
   */
  "large": PropTypes.bool,
  /**
   * use small style button
   * @examples <Button small>
   */
  "small": PropTypes.bool,

  /**
   * The class name of the icon you want to set.
   */
  "icon": PropTypes.string,

  /**
   * Pass inline styles here.
   */
  "style": PropTypes.object
};

Button.defaultProps = {
  "children": "Default Button",
  "outline": false,
  "primary": false,
  "secondary": false,
  "warning": false,
  "error": false,
  "link": false,
  "large": false,
  "small": false,
  "disabled": false,
  "type": "button"
};

export default Button;
