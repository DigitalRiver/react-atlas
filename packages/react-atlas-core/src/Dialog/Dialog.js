import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";

class Dialog extends React.PureComponent {
  _getButtonContent = () => {
    const { info, warning, onOk, onCancel, style } = this.props;
    const okButton = warning
      ? <Button small warning onClick={onOk}>
          OK
        </Button>
      : <Button small primary onClick={onOk}>
          OK
        </Button>;
    return (
      <div>
        {okButton}
        {!info &&
          <Button small onClick={onCancel}>
            Cancel
          </Button>}
      </div>
    );
  };

  render() {
    const {
      active,
      warning,
      className,
      children,
      styles,
      style,
      ...others,
    } = this.props;
    return (
      active &&
      <Modal className={cx(className)} style={style} active={active} className={warning && styles.warning} {...others}>
        <div styleName="dialog">
          {children}
          <div styleName="buttons">
            {this._getButtonContent()}
          </div>
        </div>
      </Modal>
    );
  }
}

Dialog.propTypes = {
  /**
   * Dialog's title
   */
  title: PropTypes.string,
  /** An Object, array, or string of CSS classes to apply to Dialog.*/
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),
  /**
   * Anything that can be in a Dialog.
   */
  children: PropTypes.node.isRequired,
  /**
   * Determines show Dialog or not
   */
  active: PropTypes.bool,
  /**
   * Info type
   */
  info: PropTypes.bool,
  /**
   * Confirm type with ok/cancel button
   */
  confirm: PropTypes.bool,
  /**
   * Warning type with ok/cancel button
   */
  warning: PropTypes.bool,
  /**
   * Determines show Dialog with overlay or not
   */
  overlay: PropTypes.bool,
  /**
   * Determines to hide page scroll
   */
  lockScroll: PropTypes.bool,
  /**
   * Callback for OK Button
   */
  onOk: PropTypes.func,
  /**
   * Callback for Cancel Button
   */
  onCancel: PropTypes.func,

  /* Pass inline styling here. */
  style: PropTypes.object
};

Dialog.defaultProps = {
  active: false,
  className: "",
  overlay: false,
  info: false
};

export default Dialog;
