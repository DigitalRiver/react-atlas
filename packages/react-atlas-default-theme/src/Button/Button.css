@import '../styles/variables.css';

/*Main Styles*/
.base {
  composes: button-marg-1  from '../styles.css';
  composes: button-pad-1 from '../styles.css';
  composes: default-text from '../styles.css';
  composes: cursor-pointer from '../styles.css';
  composes: primary-button-border-width from '../styles.css';
  composes: default-font from '../styles.css';
  composes: rounded from '../styles.css';
  text-decoration: none;
  display: inline-block;
  vertical-align: middle;
  line-height: 1.125rem;
  -webkit-appearance: none;
  font-size: 1.0625rem;
  border-style: solid;
}

.uppercase {
  text-transform: uppercase;
}

.base:hover {
  text-decoration: none;
}

.default_btn {
  composes: base;
  composes: hover-bg-gray from '../styles.css';
  composes: hover-white from '../styles.css';
  composes: hover-border-default from '../styles.css';
}
.default_btn:active {
  box-shadow: inset 0 0 0 20rem var(--darken-2),
  inset 0 3px 4px 0 var(--darken-3),
  0 0 1px var(--darken-2);
}
.default_btn:disabled {
  box-shadow: inset 0 0 0 20rem var(--darken-1);
}

.outline {
  composes: base;
  composes: bg-transparent from '../styles.css';
}
.outline:active {
  box-shadow: inset 0 0 0 20rem var(--darken-2),
  inset 0 3px 4px 0 var(--darken-3),
  0 0 1px var(--darken-2);
}
.outline:disabled {
  opacity: .5;
}

/*default button variant*/
.primary {
  composes: bg-primary from '../styles.css';
  composes: hover-bg-primary from '../styles.css';
  composes: border-primary from '../styles.css';
  composes: hover-border-primary from '../styles.css';
  composes: white from '../styles.css';
}
.button {
  composes: base;
}
.secondary {
  composes: bg-secondary from '../styles.css';
  composes: hover-bg-secondary from '../styles.css';
  composes: border-secondary from '../styles.css';
  composes: hover-boder-secondary from '../styles.css';
  composes: inverse-blue-text from '../styles.css';
}

/* TODO: Remove hardcoded color values and
 * compose from styles.css */
.warning {
  composes: base;
  color: #FFFFFF;
  border-color: #F07822;
  background-color: #F07822;
}

/* TODO: Remove hardcoded color values and
 * compose from styles.css */
.warning:hover {
  color: #FFFFFF;
  border-color: #E24024;
  background-color: #E24024;
}

/* TODO: Remove hardcoded color values and
 * compose from styles.css */
.error {
  composes: base;
  color: #FFFFFF;
  border-color: #EA0F43;
  background-color: #EA0F43;
}

/* TODO: Remove hardcoded color values and
 * compose from styles.css */
.error:hover {
  color: #FFFFFF;
  border-color: #BD0A33;
  background-color: #BD0A33;
}

.link {
  composes: base;
  composes: border-none from '../styles.css';
  composes: sky-blue from '../styles.css';
  composes: border-transparent from '../styles.css';
  composes: bg-transparent from '../styles.css';
}
.link:hover{
  text-decoration: underline;
  border-color: transparent;
  background-color: transparent;
  box-shadow: none;
}

/*Outline styles*/
.default_btn_outline {
  composes:outline;
  composes: hover-bg-gray from '../styles.css';
  composes: hover-white from '../styles.css';
  composes: hover-border-default from '../styles.css';
}
.primary_outline{
  composes: outline;
  composes: sky-blue from '../styles.css';
  composes: hover-bg-primary from '../styles.css';
  composes: hover-white from '../styles.css';
  composes: border-primary from '../styles.css';
  composes: hover-border-primary from '../styles.css';
}
.button_outline {
  composes: outline;
  composes: hover-bg-gray from '../styles.css';
  composes: hover-border-default from '../styles.css';
  composes: hover-white from '../styles.css';
}
.secondary_outline {
  composes: outline;
  composes: blue from '../styles.css';
  composes: hover-bg-secondary from '../styles.css';
  composes: border-secondary from '../styles.css';
  composes: hover-white from '../styles.css';
  composes: hover-border-secondary from '../styles.css';
}

.success_outline {
  composes: outline;
  composes: green from '../styles.css';
  composes: hover-bg-green from '../styles.css';
  composes: hover-white from '../styles.css';
  composes: border-green from '../styles.css';
}

.warning_outline {
  composes: outline;
  composes: orange from '../styles.css';
  composes: hover-bg-orange from '../styles.css';
  composes: hover-white from '../styles.css';
  composes: border-orange from '../styles.css';
}

.error_outline {
  composes: outline;
  composes: red from '../styles.css';
  composes: hover-bg-red from '../styles.css';
  composes: hover-white from '../styles.css';
  composes: border-red from '../styles.css';
}

.link_outline {
  composes: outline;
  composes: primary-text from '../styles.css';
  composes: bg-transparent from '../styles.css';
  composes: border-none from '../styles.css';
}
.link_outline:hover {
  text-decoration:underline;
}
.disabled {
  composes: cursor-not-allowed from '../styles.css';
  text-decoration: none;
  opacity: .65;
}

/*button sizes*/
.large {
  padding: 1rem 1.25rem;
  font-size: 1.25rem;
}

.small {
  padding: .25rem .5rem;
  font-size: .85rem;
}

/*block style*/
.block {
  display: block;
  width: 100%;
}

.icon-left {
  margin-right: 1rem;
}