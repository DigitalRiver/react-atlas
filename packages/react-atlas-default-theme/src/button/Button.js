import { Button } from "react-atlas-core";
import style from "./button.css";
import createComponent from "../createComponent";

export default createComponent(Button, style);
