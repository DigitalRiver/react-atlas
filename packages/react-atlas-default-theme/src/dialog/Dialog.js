import { Dialog } from "react-atlas-core";
import style from "./dialog.css";
import createComponent from "../createComponent";

export default createComponent(Dialog, style);
