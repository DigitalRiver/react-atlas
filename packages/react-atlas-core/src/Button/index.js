import { default as ButtonCore } from "./Button";
export { ButtonCore };
