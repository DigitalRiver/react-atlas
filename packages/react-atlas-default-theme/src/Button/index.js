import * as ButtonStyle from "./Button.css";
export { ButtonStyle };
