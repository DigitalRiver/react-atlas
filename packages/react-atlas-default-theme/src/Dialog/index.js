import * as DialogStyle from "./Dialog.css";
export { DialogStyle };
