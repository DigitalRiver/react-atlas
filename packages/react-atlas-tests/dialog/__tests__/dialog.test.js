import React from "react";
import { mount } from "enzyme";
import { DialogCore } from "../../../react-atlas-core/src/Dialog/index";
import { ButtonCore } from "../../../react-atlas-core/src/Button/index";

import renderer from 'react-test-renderer';

function handleToggle(){console.log('Inside handleToggle') };
function handleOk(){ console('-.-') };
function handleCancel(){ console('.-.') };
	
	
describe("Dialog component - Test correct render", () => {
  it("Test correct render", function() {  });
	const tree = renderer.create(<div> 
			<ButtonCore primary onClick={handleToggle}>Open Info Dialog</ButtonCore>
			<DialogCore 
        active={true} 
        onOk={handleOk} 
        title="Info"
        overlay
        info={true}
			>
				<div>This is Dialog example<br/>Any child components could be put here.</div>
			</DialogCore>
			</div>).toJSON();
	expect(tree).toMatchSnapshot(); 	
});


describe("Dialog component - Basic test", () => {
  
	it("Dialog component(info) - Basic test", function() {
    const dial = mount(<div> 
			<ButtonCore primary onClick={handleToggle}>Open Info Dialog</ButtonCore>
			<DialogCore 
        active={true} 
        onOk={handleOk} 
        title="Info"
        overlay
        info={true}
			>
				<div>This is Dialog example<br/>Any child components could be put here.</div>
			</DialogCore>
			</div>);
  });

	it("Dialog component(confirm) - Basic test", function() {
    const dial = mount(<div> 
			<ButtonCore primary onClick={handleToggle}>Open confirm Dialog</ButtonCore>
			<DialogCore 
        active={true} 
        onOk={handleOk} 
				onCancel={handleCancel}
        title="Confirm"
        overlay
        confirm={true}
			>
				<div>This is Dialog example<br/>Any child components could be put here.</div>
			</DialogCore>
			</div>);	
  });

	it("Dialog component(warning) - Basic test", function() {
    const dial = mount(<div> 
			<ButtonCore primary onClick={handleToggle}>Open warning Dialog</ButtonCore>
			<DialogCore 
        active={true} 
        onOk={handleOk} 
				onCancel={handleCancel}
        title="Warning"
        overlay
        warning={true}
			>
				<div>This is Dialog example<br/>Any child components could be put here.</div>
			</DialogCore>
			</div>);
  });
	
});

//dial.find('button').simulate('click');
//dial.findWhere( (n)=> {if(n.props().primary){n.simulate('click')}} );//.simulate('click');