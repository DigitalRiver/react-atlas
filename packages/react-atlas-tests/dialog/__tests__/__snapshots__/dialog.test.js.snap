// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <button
    className=""
    href={undefined}
    id={undefined}
    name={undefined}
    onClick={[Function]}
    style={undefined}
    styleName="button uppercase primary primary_hover"
    tabIndex={0}
    type="button"
  >
    Open Info Dialog
  </button>
</div>
`;
